test/controllers/users/applications_controller_test.rb

```diff
@@ -218,3 +304,17 @@ class Users::ApplicationsControllerTest < ActionController::TestCase
     end
   end
 end
+
+def assert_show_permissions_links(application:, user:, view:, edit:)
```

Comment:
With a view to making the assertion lines more succinct, we could move this method inside the test class and remove the first two parameters, accessing the `@application` and `@user` instance variables directly.

It might be nicer to have a separate "view" and "edit" assertion once the method signature's more clear, too.

Maybe even a more natural-feeling negation, too... I don't work with a lot of MiniTest, so I'm not up-to-date with any conventions here 😓 Just making it up, though, a more natural language approach might be: `assert_shows_view_permissions_link` / `assert_doesnt_show_view_permissions_link`, or sticking with a parameterised approach: `assert_shows_permissions_link(:view)` or `assert_shows_permissions_link(:view_permissions)`. Squeezing what I think might be some MiniTest lingo in here, we could try: `refute_shows_view_permissions_link`! My personal preference today is for something as concise as possible while still maintaining that natural language, like: `assert_view_permissions_link` / `assert_no_view_permissions_link` 🤷

Reply:
Something like this?

https://github.com/alphagov/signon/pull/3040/commits/ae7e413d5d14c988f26ebc6270b93350d1151fbe

Reply:
Yes 👍 The assertions in the tests are much clearer now